src/parser.rs

```diff
@@ -161,7 +161,9 @@ impl<'a> Parser<'a> {
                 return parser.expected("end of statement", parser.peek_token());
             }
 
-            let statement = parser.parse_statement()?;
+            let statement = stacker::maybe_grow(2 * 1024 * 1024, 5 * 1024 * 1024, || {
```

Comment:
So looking at this code more carefully, I am not sure exactly how it is working -- specifically this call to `maybe_grow` grows the stack if there is less then 2MB remaining

In fact the test passes even if I remove the `maybe_grow` around this call (it fails if i remove the check on the `impl Display`)

Reply:
```diff
diff --git a/src/parser.rs b/src/parser.rs
index 5fd7af2..cff2281 100644
--- a/src/parser.rs
+++ b/src/parser.rs
@@ -161,9 +161,7 @@ impl<'a> Parser<'a> {
                 return parser.expected("end of statement", parser.peek_token());
             }
 
-            let statement = stacker::maybe_grow(2 * 1024 * 1024, 5 * 1024 * 1024, || {
-                parser.parse_statement()
-            })?;
+            let statement = parser.parse_statement()?;
             stmts.push(statement);
             expecting_statement_delimiter = true;
         }
```

And then

```shell
cd /Users/alamb/Software/sqlparser-rs/ && RUST_BACKTRACE=1 CARGO_TARGET_DIR=/Users/alamb/Software/target-df cargo test -- overflow
    Finished test [unoptimized + debuginfo] target(s) in 0.04s
     Running unittests src/lib.rs (/Users/alamb/Software/target-df/debug/deps/sqlparser-d27a7d1b89bd35ed)

running 1 test
test parser::tests::test_stack_overflow ... ok
```

🤔

Reply:
+1